### PR TITLE
Easter cleanup - project wide compile warnings

### PR DESCRIPTION
**Pull request by tworec**

### CODE REVIEW

File: ethkey/src/lib.rs [hunk 1]
```diff
@@ -177,12 +173,19 @@ impl fmt::Debug for EthAccount {
...
 
-error_chain! {
-    foreign_links {
-        IoError(io::Error);
-        EthsignError(ethsign::Error);
-        Secp256k1Error(secp256k1::Error);
-        SerdeJsonError(serde_json::Error);
+mod error {
+    use error_chain::{
+        error_chain, error_chain_processing, impl_error_chain_kind, impl_error_chain_processed,
```
Comment (prekucki) [inline ethkey/src/lib.rs]: importing of internal macros. use error_chain::* is better.

File: gu-actix/src/pipe.rs [hunk 2]
```diff
@@ -36,7 +37,7 @@ impl<T, E> Stream for AsyncReader<T, E> {
 impl io::Read for SyncReader<Bytes, io::Error> {
     fn read(&mut self, buf: &mut [u8]) -> Result<usize, io::Error> {
         if self.buffer.is_none() {
-            let is_full = self.rx.is_full();
+            let _is_full = self.rx.is_full();
```
Comment (prekucki) [inline gu-actix/src/pipe.rs]: TODO: remove or add optimalization.

**Review by prekucki — CHANGES_REQUESTED** [review]

File: gu-hdman/src/cache.rs [hunk 3]
```diff
@@ -114,8 +111,8 @@ impl<P: CacheProvider> CacheRegistry<P> {
         let _ = self.downloads.remove(&key).and_then(|v| {
             for endpoint in v {
                 match endpoint.send(result.clone()) {
```
Comment (prekucki) [inline gu-hdman/src/cache.rs:112]: !

File: gu-hdman/src/download/error.rs [hunk 4]
```diff
@@ -18,7 +20,7 @@ pub enum Error {
     Serialize(#[fail(cause)] bincode::Error),
 
     #[fail(display = "Overflow")]
```
Comment (prekucki) [inline gu-hdman/src/download/error.rs]: !

File: gu-hdman/src/download/sync_io.rs [hunk 5]
```diff
@@ -260,7 +255,10 @@ impl DownloadFile {
         use crc::Hasher64;
 
         if chunk_nr >= self.meta.chunks {
-            Err(Error::Overflow)
```
Comment (prekucki) [inline gu-hdman/src/download/sync_io.rs:263]: +

File: gu-hub/src/local_service.rs [hunk 6]
```diff
@@ -223,15 +228,15 @@ enum ProxyPath {
...
 
         let mut b = client::ClientRequest::build();
 
         match self {
             ProxyPath::Remote { url } => {
                 b.uri(&format!("{}{}", url, path));
             }
-            ProxyPath::Local { cmd } => {
+            ProxyPath::Local { cmd: _ } => {
                 unimplemented!();
```
Comment (prekucki) [inline gu-hub/src/local_service.rs:240]: ```ProxyPath::Local { ... } => {```

File: gu-actix/src/pipe.rs (hunk 2 above)
Comment (tworec) [reply gu-actix/src/pipe.rs]: removed

File: gu-hdman/src/download/sync_io.rs (hunk 5 above)
Comment (tworec) [reply gu-hdman/src/download/sync_io.rs:263]: changed to `InvalidTrackingFile`

File: gu-hdman/src/cache.rs (hunk 3 above)
Comment (tworec) [reply gu-hdman/src/cache.rs:112]: reverted Debug

File: gu-hub/src/local_service.rs (hunk 6 above)
Comment (tworec) [reply gu-hub/src/local_service.rs:240]: done

File: gu-hdman/src/download/error.rs (hunk 4 above)
Comment (tworec) [reply gu-hdman/src/download/error.rs]: done

File: ethkey/src/lib.rs (hunk 1 above)
Comment (tworec) [reply ethkey/src/lib.rs]: done

**Review by prekucki — APPROVED** [review]

**Pull request merged**